commit 26270d4b231f6554d2820877cd588acdd482f1b1
Author: Ke Wu <mikewu@google.com>
Date:   Tue Nov 8 15:21:08 2016 -0800

    Catch OSError (read-only file system).
    
    _UpdateAuthorizedKeys() will set write permission and add write key to
    every user's home directory. However, the home directory of user 'root'
    (/root) is read only in lakitu.

diff --git a/google_compute_engine/accounts/accounts_utils.py b/google_compute_engine/accounts/accounts_utils.py
index 4096086..93bed21 100644
--- a/google_compute_engine/accounts/accounts_utils.py
+++ b/google_compute_engine/accounts/accounts_utils.py
@@ -161,6 +161,8 @@ class AccountsUtils(object):
 
     Raises:
       IOError, raised when there is an exception updating a file.
+      OSError, raised when try to set permission or write to a read-only
+          file system, e.g. /root in lakitu.
     """
     pw_entry = self._GetUser(user)
     if not pw_entry:
@@ -296,7 +298,7 @@ class AccountsUtils(object):
 
     try:
       self._UpdateAuthorizedKeys(user, ssh_keys)
-    except IOError as e:
+    except (IOError, OSError) as e:
       message = 'Could not update the authorized keys file for user %s. %s.'
       self.logger.warning(message, user, str(e))
       return False
